### ISSUE TITLE
Requests and Responses do not have source maps

### ISSUE THREAD

**Issue by honzajavorek** [apiaryio/drafter#259]
Currently, HTTP Requests and HTTP Responses do not have source maps in the `refract` output. Their sub-elements and attributes have source maps, but there's no way to determine where the actual `+ Request` or `+ Response` happened to be in the original document.

In following situation I can distinguish the requests, because they happen to have headers. Headers have source maps, so I can say which request was on which position:

```apib
+ Request (application/json)
+ Response 200

+ Request (application/json)
+ Response 200
```

However, imagine following situation with "bare" requests:

```apib
+ Request
+ Response 200

+ Request
+ Response 200
```

In this case the requests do not have a single sub-element with a source map (except of the `method` attribute, but that doesn't actually say where the `+ Request` was written). I can say where responses were positioned, because I have source maps for the status codes. However, there is no way to say this for the requests. I can't distinguish which was which or whether they're the same and I can't re-construct number of the implicit HTTP transaction example (in this case, the first request is Example 1, the second is Example 2).

This fact blocks me to continue with https://github.com/apiaryio/dredd-transactions/pull/2. I can't flawlessly implement backward compatibility for https://github.com/apiaryio/dredd/issues/388 and I can't support [API Elements](http://api-elements.readthedocs.org/en/latest/overview/) in Dredd.

**Comment by honzajavorek** [comment]
By trial and error, I found out the following cases won't allow me to reconstruct the original document. While the combinations are not often very useful on their own and sometimes they produce warnings, they're still non-erroneous API Blueprint notation which could appear in Dredd.

### Problematic Cases

#### Bare Request

```apib
+ Request
+ Response 200
```

#### Request with Bare Attributes Section

```apib
+ Request
    + Attributes
+ Response 200
```

#### Request with Bare Body Section

```apib
+ Request
    + Body
+ Response 200
```

#### Request with Bare Schema Section

```apib
+ Request
    + Schema
+ Response 200
```

#### Bare Response

```apib
+ Request (application/json)
+ Response
```

#### Response with Bare Attributes Section

```apib
+ Request (application/json)
+ Response
    + Attributes
```

#### Response with Bare Body Section

```apib
+ Request (application/json)
+ Response
    + Body
```

#### Response with Bare Schema Section

```apib
+ Request (application/json)
+ Response
    + Schema
```

**Comment by honzajavorek** [comment]
I can confirm this has been resolved in `protagonist@1.3.0-pre.1` :tada:
